scripts/pi-hole/js/utils.js

```diff
@@ -248,6 +248,81 @@ function getGraphType() {
   return localStorage.getItem("barchart_chkbox") === "false" ? "line" : "bar";
 }
 
+function addFromQueryLog(domain, list) {
+  var token = $("#token").text();
+  var alertModal = $("#alertModal");
+  var alProcessing = alertModal.find(".alProcessing");
+  var alSuccess = alertModal.find(".alSuccess");
+  var alFailure = alertModal.find(".alFailure");
+  var alNetworkErr = alertModal.find(".alFailure #alNetErr");
+  var alCustomErr = alertModal.find(".alFailure #alCustomErr");
+  var alList = "#alList";
+  var alDomain = "#alDomain";
+
+  // Exit the function here if the Modal is already shown (multiple running interlock)
+  if (alertModal.css("display") !== "none") {
+    return;
+  }
+
+  var listtype = list === "white" ? "Whitelist" : "Blacklist";
+
+  alProcessing.children(alDomain).html(domain);
+  alProcessing.children(alList).html(listtype);
+  alertModal.modal("show");
+
+  // add Domain to List after Modal has faded in
+  alertModal.one("shown.bs.modal", function () {
+    $.ajax({
+      url: "scripts/pi-hole/php/groups.php",
+      method: "post",
+      data: {
+        domain: domain,
+        list: list,
+        token: token,
+        action: "replace_domain",
+        comment: "Added from Query Log"
+      },
+      success: function (response) {
+        alProcessing.hide();
+        if (!response.success) {
+          // Failure
+          alNetworkErr.hide();
+          alCustomErr.html(response.message);
+          alFailure.fadeIn(1000);
+          setTimeout(function () {
+            alertModal.modal("hide");
+          }, 10000);
+        } else {
+          // Success
+          alSuccess.children(alDomain).html(domain);
+          alSuccess.children(alList).html(listtype);
+          alSuccess.fadeIn(1000);
+          setTimeout(function () {
+            alertModal.modal("hide");
+          }, 2000);
+        }
+      },
+      error: function () {
+        // Network Error
+        alProcessing.hide();
+        alNetworkErr.show();
+        alFailure.fadeIn(1000);
+        setTimeout(function () {
+          alertModal.modal("hide");
+        }, 8000);
```

Comment:
and/or 8 seconds might also be too long still.

Maybe 5?

Reply:
5 instead of 4 doesn't really sound like much of a difference, though?

Reply:
I added a close button.